cli/internal/otel/receiver.go

```diff
@@ -0,0 +1,358 @@
+package otel
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+	"go.uber.org/zap"
+	"golang.org/x/exp/maps"
+)
+
+type Span struct {
+	TraceID        string         `json:"trace_id"`
+	SpanID         string         `json:"span_id"`
+	ParentSpanID   string         `json:"parent_span_id"`
+	Name           string         `json:"name"`
+	StartTimestamp time.Time      `json:"start_timestamp"`
+	EndTimestamp   time.Time      `json:"end_timestamp"`
+	Attributes     map[string]any `json:"attributes"`
+}
+
+type Metric struct {
+	Name        string         `json:"name"`
+	Description string         `json:"description"`
+	Unit        string         `json:"unit"`
+	Value       int64          `json:"value"`
+	Attributes  map[string]any `json:"attributes"`
+}
+
+type TableMetric struct {
+	Table     string     `json:"table"`
+	ClientId  string     `json:"client_id"`
+	StartTime *time.Time `json:"start_time"`
+	EndTime   *time.Time `json:"end_time"`
+	Resources int64      `json:"resources"`
+	Errors    int64      `json:"errors"`
+	Panics    int64      `json:"panics"`
+}
+
+type ConsumeSpan = func(context.Context, Span)
+
+type ConsumeMetric = func(context.Context, Metric)
+
+type Consumer struct {
+	consumeSpan     ConsumeSpan
+	quit            chan any
+	metricsFilename string
+	metricsFile     *os.File
+	consumeMetric   ConsumeMetric
+}
+
+func (c *Consumer) Shutdown(ctx context.Context) error {
+	var err error
+	if c.quit != nil {
+		close(c.quit)
+	}
+	if c.metricsFile != nil {
+		err = errors.Join(err, c.metricsFile.Close())
```

Comment:
This joins ` c.metricsFile.Close()` with `nil`. I assume it was a rewrite leftover?

Reply:
Fixed in https://github.com/cloudquery/cloudquery/pull/18397/commits/0f79d15bac686b591c2de5e0192d9a42075fe349